2006-02-12  Ulrich Drepper  <drepper@redhat.com>

	[BZ #2226]
	* libio/wgenops.c (_IO_wsetb): Use correct size of wide char
	buffer in FREE_BUF call.

--- libc/libio/wgenops.c	14 Sep 2004 04:24:45 -0000	1.13
+++ libc/libio/wgenops.c	12 Feb 2006 21:20:57 -0000	1.14
@@ -116,7 +116,7 @@ _IO_wsetb (f, b, eb, a)
      int a;
 {
   if (f->_wide_data->_IO_buf_base && !(f->_flags & _IO_USER_BUF))
-    FREE_BUF (f->_wide_data->_IO_buf_base, _IO_wblen (f));
+    FREE_BUF (f->_wide_data->_IO_buf_base, _IO_wblen (f) * sizeof (wchar_t));
   f->_wide_data->_IO_buf_base = b;
   f->_wide_data->_IO_buf_end = eb;
   if (a)
